File: pset2/Caesar/caesar.c
# include <stdio.h>
# include <cs50.h>
# include <stdlib.h>
# include <string.h>
# include <ctype.h>

int main(int argc, string argv[])
{
    if (argc == 1)
    {
        return 1;
    }
    int key = atoi(argv[1]);

    string plainText = get_string("plaintext:  ");
    if (!plainText)
    {
        return 1;
    }

    for (int i = 0, n = strlen(plainText); i < n; i++)
    {
        if (isalpha(plainText[i]))
        {
            if (isupper(plainText[i]))
            {
                int num = (int)plainText[i] - 65;
                num = ((num + key) % 26) + 65;
                plainText[i] = num;
            }
            else
            {
                int num = (int)plainText[i] - 97;
                num = ((num + key) % 26) + 97;
                plainText[i] = num;
            }
        }
        else
        {
            int num = (int)plainText[i];
            plainText[i] = num;
        }
    }

    printf("ciphertext: %s\n", plainText);
}

File: pset6/caesar/caesar.py
from cs50 import get_string
import sys


def main():
    if len(sys.argv) != 2:
        sys.exit(1)

    num = 0

    key = int(sys.argv[1])

    plainText = get_string("plainText: ")
    print("ciphertext: ", end="")
    for i in range(0, len(plainText)):
        if plainText[i].isalpha():
            if plainText[i].isupper():
                num = ord(plainText[i]) - 65
                num = ((num + key) % 26) + 65
                print(chr(num), end="")
            else:
                num = ord(plainText[i]) - 97
                num = ((num + key) % 26) + 97
                print(chr(num), end="")
        else:
            print(plainText[i], end="")
    print("")


if __name__ == "__main__":
    main()